Mathlib/Topology/Algebra/Module/WeakDual.lean

```diff
@@ -314,6 +314,45 @@ instance instTopologicalSpace : TopologicalSpace (WeakSpace 𝕜 E) :=
 instance instContinuousAdd : ContinuousAdd (WeakSpace 𝕜 E) :=
   WeakBilin.instContinuousAdd (topDualPairing 𝕜 E).flip
 
+
+/-- There is a canonical map `E → WeakSpace 𝕜 E` (the "identity"
+mapping). It is a linear equivalence. -/
+def toWeakSpace : E ≃ₗ[𝕜] WeakSpace 𝕜 E := LinearEquiv.refl 𝕜 E
+-- align?
+
+@[simp]
+-- theorem coe_toWeakSpace (x : E) : (toWeakSpace x : WeakSpace 𝕜 E) = x :=
+theorem coe_toWeakSpace (x : E) : toWeakSpace x = x :=
+  rfl
+-- #align ?
+
+@[simp]
+theorem toWeakSpace_eq_iff (x y : E) : (toWeakSpace x : WeakSpace 𝕜 E) = toWeakSpace y ↔ x = y :=
+  Function.Injective.eq_iff <| LinearEquiv.injective toWeakSpace
+-- #align ?
+
+-- theorem toWeakSpace_continuous : Continuous fun x : E => (toWeakSpace x : WeakSpace 𝕜 E) := by
+theorem toWeakSpace_continuous : Continuous fun x : E => toWeakSpace x := by
+  refine { isOpen_preimage := ?isOpen_preimage }
+  intro U hU
+  apply Continuous.le_induced _ U hU
+  apply continuous_pi_iff.mpr
+  intro y
+  exact ContinuousLinearMap.continuous y
+
+
+/-- For a topological vector space `E`, according to `toWeakSpace_continuous` the "identity mapping"
+`E → WeakSpace 𝕜 E` is continuous. This definition implements it as a continuous linear map. -/
+def continuousLinearMapToWeakSpace : E →L[𝕜] WeakDual 𝕜 E :=
+  { toWeakSpace with cont := toWeakSpace_continuous }
+-- #align ?
+
+/-- There is a canonical map `E → WeakSpace 𝕜 E` (the "identity"
+mapping). It is a linear equivalence. -/
+def toOriginalSpace : WeakSpace 𝕜 E ≃ₗ[𝕜] E := LinearEquiv.refl 𝕜 E
+-- align?
```

Comment:
I think we don't need this as it is exactly `toWeakSpace.symm`

Reply:
here again, `NormedSpace.WeakDual` has [WeakDual.toNormedDual](https://leanprover-community.github.io/mathlib4_docs/Mathlib/Analysis/NormedSpace/WeakDual.html#WeakDual.toNormedDual). Which is the current convention?

Reply:
Yeah, `NormedSpace.WeakDual` needs some serious clean-up, I will take care of that unless you want to do it.

Reply:
ok, I deleted this theorem. I leave `NormedSpace` to you (or others).